Sources/PromiseKit/Promise.swift

```diff
@@ -176,8 +178,30 @@ public extension Promise {
     `catch`(work)
     return self
   }
+  
+  /// Sleep promise execution for given `duration` interval and return new promise with existing value.
+  func sleep(
+    duration: DispatchTimeInterval,
+    on dispatchQueue: DispatchQueue = .main
+  ) -> Promise<Value> {
+    Promise { seal in
+      self.finally {
+        switch $0 {
+        case .success(let value):
+          dispatchQueue.asyncAfter(deadline: .now() + duration) {
+            seal.resolve(with: value)
```

Comment:
Resolve the promise on the provided queue.

Reply:
https://github.com/poviolabs/PovioKit/pull/170/commits/6bbec698589d7d97ae735f331dbbfaa881698a1f